### ISSUE TITLE
getInfo method should not reject before opening the port

### ISSUE THREAD

**Issue by stylesuxx** [google/web-serial-polyfill#19]
In the "real" Web Serial API one can query information about the available ports before opening a connection:

```javascript
const ports = await serial.getPorts();
const info = await ports[0].getInfo();
```
But results in a rejection with the polyfill:

```
DOMException: The device must be opened first
```

**Comment by reillyeon** [comment]
The current implementation of `getInfo()` actually doesn't match the specification at all. It returns a [`SerialOptions`](https://wicg.github.io/serial/#serialoptions-dictionary) rather than a [`SerialPortInfo`](https://wicg.github.io/serial/#serialportinfo-dictionary). That's why it requires the device to be open, so it can read back the current device configuration. If it returned a `SerialPortInfo` that could be constructed only from attributes of the underlying `USBDevice` interface.

**Comment by stylesuxx** [comment]
I understand. But in order to be closer to the Web Serial implementation, it should not reject. Otherwise I would need to again have checks in place if the polyfill or the real deal is used - which is not what I should be having to do when using a polyfill.

If device info is not available would it not be an option to simply return some dummy data for vendor and product id?

**Comment by reillyeon** [comment]
This was fixed in #21.
